x-pack/legacy/plugins/ml/public/data_frame/pages/job_management/components/job_list/preview_pane.tsx

```diff
@@ -0,0 +1,166 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiInMemoryTable, EuiInMemoryTableProps, SortDirection } from '@elastic/eui';
+import { getFlattenedFields } from '../../../../components/source_index_preview/common';
+import { ml } from '../../../../../services/ml_api_service';
+
+import { DataFramePreviewRequest } from '../../../../common';
+
+interface Transform {
+  dest: { index: string };
+  id: string;
+  pivot: {
+    group_by: { [key: string]: any };
+    aggregations: { [key: string]: any };
+  };
+  source: {
+    index: string[];
+    query: { [key: string]: any };
+  };
+}
+
+interface TransformData {
+  count: number;
+  transforms: Transform[];
+}
+
+interface CompressedTableProps extends EuiInMemoryTableProps {
+  compressed: boolean;
+  error: any;
+}
+
+interface Column {
+  field: string;
+  name: string;
+  sortable: boolean;
+  truncateText: boolean;
+}
+
+interface Props {
+  lastUpdate: number;
+  transformId: string;
+}
+
+export function sortColumns(groupByArr: string[]) {
+  return (a: string, b: string) => {
+    // make sure groupBy fields are always most left columns
+    if (groupByArr.some(aggName => aggName === a) && groupByArr.some(aggName => aggName === b)) {
+      return a.localeCompare(b);
+    }
+    if (groupByArr.some(aggName => aggName === a)) {
+      return -1;
+    }
+    if (groupByArr.some(aggName => aggName === b)) {
+      return 1;
+    }
+    return a.localeCompare(b);
+  };
+}
+
+function getDataFromTransform(
+  transformData: TransformData
+): { previewRequest: DataFramePreviewRequest; groupByArr: string[] | [] } {
+  const { transforms } = transformData;
+  const index = transforms[0].source.index[0];
+  const pivot = transforms[0].pivot;
+  const groupByArr = [];
+
+  const previewRequest: DataFramePreviewRequest = {
+    source: {
+      index,
+    },
+    pivot,
+  };
+  // hasOwnProperty check to ensure only properties on object itself, and not its prototypes
+  if (pivot.group_by !== undefined) {
+    for (const key in pivot.group_by) {
+      if (pivot.group_by.hasOwnProperty(key)) {
+        groupByArr.push(key);
+      }
+    }
+  }
+
+  return { groupByArr, previewRequest };
+}
+
+export const PreviewPane: React.SFC<Props> = ({ transformId, lastUpdate }) => {
+  const [dataFramePreviewData, setDataFramePreviewData] = useState([]);
+  const [groupByArray, setGroupByArray] = useState<string[] | []>([]);
+  const [columns, setColumns] = useState<Column[] | []>([]);
+  const [sort, setSort] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  async function getPreview() {
+    try {
+      const transformData: any = await ml.dataFrame.getDataFrameTransforms(transformId);
```

Comment:
Maybe we can pass on the transform config as a prop and don't need to fetch it again? It's available as `.config` on the `DataFrameJobListRow` objects.